go-controller/pkg/cluster/node.go

```diff
@@ -136,10 +148,6 @@ func (cluster *OvnClusterController) StartClusterNode(name string) error {
 
 	// First wait for the node logical switch to be created by the Master, timeout is 300s.
 	err = wait.PollImmediate(500*time.Millisecond, 300*time.Second, func() (bool, error) {
-		if node, err = cluster.Kube.GetNode(name); err != nil {
-			logrus.Errorf("error retrieving node %s: %v", name, err)
-			return false, nil
-		}
 		cidr, err = getNodeHostSubnetAnnotation(node)
```

Comment:
We need the above statements. Otherwise, we will be looking for annotation on the already retrieved Node object. If the node was not annotated with subnet information @L128, then this re-try will just fail after 300 seconds. So, we need this:

```
	if node, err = cluster.Kube.GetNode(name); err != nil {
		return fmt.Errorf("error retrieving node %s: %v", name, err)
	}
	err = setupOVNNode(node)
	if err != nil {
		return err
	}

	for _, clusterSubnet := range config.Default.ClusterSubnets {
		clusterSubnets = append(clusterSubnets, clusterSubnet.CIDR.String())
	}

	messages := make(chan error)

	// First wait for the node logical switch to be created by the Master, timeout is 300s.
	err = wait.PollImmediate(500*time.Millisecond, 300*time.Second, func() (bool, error) {
		if node, err = cluster.Kube.GetNode(name); err != nil {
			logrus.Errorf("error retrieving node %s: %v", name, err)
			return false, nil
		}
		cidr, err = getNodeHostSubnetAnnotation(node)
		if err != nil {
			logrus.Errorf("Error starting node %s, no annotation found on node for subnet - %v", name, err)
			return false, nil
		}
		return true, nil
	})
	if err != nil {
		return fmt.Errorf("timed out waiting for node's: %q logical switch: %v", name, err)
	}
```

Reply:
What was I thinking!?